src/contracts/core/DelegationManager.sol

```diff
@@ -293,6 +300,24 @@ contract DelegationManager is Initializable, OwnableUpgradeable, Pausable, Deleg
         // retrieve `staker`'s list of strategies and the staker's shares in each strategy from the StrategyManager
         (IStrategy[] memory strategies, uint256[] memory shares) = strategyManager.getDeposits(staker);
 
+        uint256 beaconChainETHShares = eigenPodManager.getBeaconChainETHShares(staker);
+        IStrategy beaconChainETHStrategy = eigenPodManager.beaconChainETHStrategy();
+        if(beaconChainETHShares > 0) {
+            uint256 numStrategies = strategies.length;
+            IStrategy[] memory newStrategies = new IStrategy[](numStrategies + 1);
+            uint256[] memory newShares = new uint256[](numStrategies + 1);
+            // add beaconChainETH shares to the staker's shares
+            for (uint256 i = 0; i < numStrategies; i++) {
+                if(i == numStrategies - 1) {
+                    newStrategies[i] = beaconChainETHStrategy;
+                    newShares[i] = beaconChainETHShares;
+                } else {
+                    newStrategies[i] = strategies[i];
+                    newShares[i] = shares[i];
+                }
+            }
+        }
```

Comment:
these memory operations seem quite inefficient -- couldn't we just have a single step outside of the other loop to add these shares?

Reply:
yeah idk wtf I was thinking, fixed